packages/api/authentication/preprocessor.go

```diff
@@ -0,0 +1,47 @@
+package authentication
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/dopedao/dope-monorepo/packages/api/ent"
+	"github.com/dopedao/dope-monorepo/packages/api/ent/wallet"
+	"github.com/jiulongw/siwe-go"
+)
+
+type authWalletCtxKey struct{}
+
+// preprocess all upcoming requests
+func PreProcessor(client *ent.Client) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			session, err := Store.Get(r, "session")
+			// we could check if the session is new to know
+			// if the user is authenticated or not but lets check
+			// if siwe signed message is present instead
+			if err != nil || session.Values["siwe"] == nil {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			// get wallet data
+			siweMessage := session.Values["siwe"].(siwe.Message)
```

Comment:
we can serialize and store the wallet entity in the cookie, to prevent doing a db lookup on every request